tests/publicsuffixlist.js

```diff
@@ -24,16 +25,88 @@ process.on('warning', warning => {
     }
 });
 
+class MockWindow {
+    constructor() {
+        this.CustomEvent = class {
+            constructor(name, detail) {
+                this.name = name;
+
+                if ( typeof detail !== 'undefined' ) {
+                    this.detail = detail;
+                }
+            }
+        };
+
+        this.eventLog = [];
+    }
+
+    dispatchEvent(event) {
+        this.eventLog.push(event);
+    }
+}
+
+function prepareGlobals() {
+    const window = new MockWindow();
+    const { CustomEvent } = window;
+
+    return { Object, Function, window, CustomEvent };
+}
+
 describe('publicsuffixlist', () => {
     let psl = null;
+    let eventLog = null;
 
-    beforeEach(async () => {
-        psl = (await createWorld('./publicsuffixlist.js')).default;
+    afterEach(() => {
+        eventLog = null;
+        psl = null;
     });
 
     describe('version', () => {
+        beforeEach(async () => {
+            psl = (await createWorld('./publicsuffixlist.js')).default;
+        });
+
         it('should be 3.0', () => {
-            assert.equal(psl.version, '3.0');
+            assert.strictEqual(psl.version, '3.0');
+        });
+    });
+
+    describe('parse()', () => {
+        context('Mock browser', () => {
+            beforeEach(async () => {
```

Comment:
These tests are only for the `publicSuffixListChanged` event in a browser environment.